js/web/web-backend.js

```diff
@@ -55,7 +55,7 @@ async function withThreadPool(run) {
   if (workerPromise === undefined)
     throw Error('need to initialize worker first');
   let worker = await workerPromise;
-  numWorkers ??= await getEfficientNumWorkers();
+  numWorkers ??= workers.numWorkers ?? navigator.hardwareConcurrency;
```

Comment:
here as well, `navigator.hardwareConcurrency - 1` would make sense to me personally, although it probably won't make much of a difference (on the intel CPUs that I have, benefits from parallelism steeply diminish after the 4th thread or so)